Font size changing in editor help section

siomnnn opened the pull request:
Made font size changing work in the editor's help section, which was previously not affected by `main_font_size`. Fixes #35269 .

siomnnn committed:
Font size changing in editor help

Tried to make font size changing work in the editor help, which was previously not affected by main_font_size.

akien-mga commented:
Well...

As you can see from the code you're removing, there's actually a dedicated option for setting the help font sizes. With this change you're removing its use, thus making the still existing options inactive.

noname1477 commented:
@johnnygossdev suggested that I should not introduce a new option to the settings, but rather scale it with `main_font_size`. Previously, you couldn't change the scale at all (without doing it from code).

noname1477 commented:
Given that we currently have 6 `Font Size` settings (1 basic font, 1 for the code editor, 3 for the help section, 1 for the output) in 3 different places, I would prefer grouping everything font related as a new `Font` tab. Currently, it's just confusing.

Calinou commented:
We can probably remove all the help font size settings and multiply the main font size instead.

johnnygossdev commented:
Ah I see the different fonts size options. I should have done a search - I just expected the fonts to be in the same place :sweat_smile:

siomnnn committed:
Remove useless settings

In my first commit, I made the help font sizes dependent on the main font size setting, but forgot to remove the now useless settings to change them.

noname1477 commented:
As suggested by @johnnygossdev and @Calinou I simply removed the useless settings.

Calinou commented:
Some files were moved to the wrong directories, `editor_settings.cpp` should be in `editor/`, not in the root directory (same for some other files).

noname1477 commented:
Oh, sorry, I didn't see that. I'm still pretty new to GitHub, so I do this mistake all the time.

siomnnn committed:
Revert "Remove useless settings"

This reverts commit 53898405412c456b3d0902203704d492377c5085.

siomnnn committed:
Remove useless settings, 2nd try

Calinou commented on editor/editor_fonts.cpp:
```diff
@@ -243,10 +243,10 @@ void editor_register_fonts(Ref<Theme> p_theme) {
-	MAKE_SOURCE_FONT(df_doc_code, int(EDITOR_GET("text_editor/help/help_source_font_size")) * EDSCALE);
+	MAKE_DEFAULT_FONT(df_doc, (default_font_size + 1) * EDSCALE);
+	MAKE_BOLD_FONT(df_doc_bold, (default_font_size + 1) * EDSCALE);
+	MAKE_BOLD_FONT(df_doc_title, (2 * default_font_size - 1) * EDSCALE);
```
Won't this set the title font size to 27 where it was previously 23? 27 may be a bit too large.

Other than that, this PR looks good to me now.

YuriSizov commented:
Superseded by https://github.com/godotengine/godot/pull/73106.
